core/base/topoMap/TopoMap.h

```diff
@@ -0,0 +1,1045 @@
+          T coordARotate[2] = {coords1Test[2 * i], coords1Test[2 * i + 1]};
+          for(size_t j = 0; j < comp2Size; j++) {
+            T coordBRotate[2] = {coords2Test[2 * j], coords2Test[2 * j + 1]};
+            T newDist = compute_dist(coordARotate, coordBRotate);
+            curScore += (newDist - origDistMatrix[i][j])
+                        * (newDist - origDistMatrix[i][j]);
+            if(newDist + Epsilon < shortestDistPossible) {
+              curAngleError = true;
+              errorDuringLoop = true;
+              break;
+            }
+          }
+        }
+
+        if(curAngleError) {
+          continue;
+        }
+
+        // Now if the score is the best so far, we keep this pair angle as the
+        // best one we tried so far.
+#ifdef TTK_ENABLE_OPENMP
+#pragma omp critical
+#endif
+        {
+          // In case of errors in convex hull computation, we still have a valid
+          // angle (to preserve the MST).
+          foundValidAngle = true;
+          // This pair of angle minimises the distortion and the choice is
+          // deterministic with threads.
+          if(curScore < bestScore
+             || (curScore == bestScore
+                 && (testAngle1 < bestAnglePair[0]
+                     || (testAngle1 == bestAnglePair[0]
+                         && testAngle2 < bestAnglePair[1])))) {
+            bestScore = curScore;
+            bestAnglePair[0] = std::isfinite(testAngle1) ? testAngle1 : 0;
+            bestAnglePair[1] = std::isfinite(testAngle2) ? testAngle2 : 0;
+          }
+        }
+      }
+    }
+
+    // No valid angle were found, because the convex hull is invalid;
+    if(!foundValidAngle) {
+      this->printErr(
+        "No valid angle was found, due to errors in the convex hull "
+        "computation. Please consider using Qhull instead of Boost. Aborting.");
+      return -10;
+    }
+
+    // Applying the chosen rotations to the two components.
+    for(size_t i1 : comp1) {
+      rotate(&allCoords[2 * i1], coordPt1, bestAnglePair[0]);
+    }
+    for(size_t i2 : comp2) {
+      rotate(&allCoords[2 * i2], coordPt2, bestAnglePair[1]);
+    }
+
+    if(errorDuringLoop) {
+      this->errorConvexHull = true;
+    }
+
+    return bestScore;
+  }
+
+  template <typename T>
+  bool TopoMap::computeConvexHull(T *allCoords,
+                                  const std::vector<size_t> &compPtsIds,
+                                  std::vector<size_t> &idsInHull) const {
+    size_t nbPoint = compPtsIds.size();
+    std::vector<double> compCoords(2 * nbPoint);
+    for(size_t i = 0; i < nbPoint; i++) {
+      size_t vertId = compPtsIds[i];
+      compCoords[2 * i] = static_cast<double>(allCoords[2 * vertId]);
+      compCoords[2 * i + 1] = static_cast<double>(allCoords[2 * vertId + 1]);
+    }
+
+    if(nbPoint <= 2) {
+      idsInHull.push_back(compPtsIds[0]);
+      if(nbPoint == 2) {
+        double dist = compute_dist(&compCoords[0], &compCoords[2]);
```

Comment:
please use instead `ttk::Geometry::distance(ptA, ptB, 2);`

Reply:
I changed it, I will try to see next week if there is an impact on performances (reimplementing the function did not use a loop, hence no branching.)

Reply:
For "big" number of angle intervals, the performance is much slower if calling distance, hence a new distance2D function.